neqo-qpack/src/encoder.rs

```diff
@@ -359,6 +358,20 @@ impl QPackEncoder {
         stream_id: u64,
     ) -> Res<HeaderEncoder> {
         qdebug!([self], "encoding headers.");
+
```

Comment:
This looks good.  I can see where this might interleave instructions.  But I couldn't work out which test case would have caused this to break.

Reply:
It is test zero_rtt_send_request. To trigger the bug:
- 0rtt needs to be used(so we have a capacity change waiting to be sent. Without 0rtt we usually send the capacity change before we try to send a request, but the bug may be triggered in that code path as well)
- a header that is not in dynamic or static table need to be present in a request, e.g. "newheader: newvalue".
In test zero_rtt_send_request I have added &[(String::from("myheaders"), String::from("myvalue"))].